ReportCardParser/src/reportcardparser/ReportCardParserType2.java

```diff
@@ -5,32 +5,96 @@
  */
 package reportcardparser;
 
+import java.io.*;
+import java.util.*;
+
 /**
  * Class to handle parsing of a Type 2 report card
  * 
  * @author chris
  */
-public class ReportCardParserType2 implements IReportCardParser {
-    
-    private String filePath; // Path to the report card file
+public class ReportCardParserType2 extends AbstractReportCardParser {
     
     /**
      * Constructor
      * 
      * @param filePath Path to the report card file
      */
     public ReportCardParserType2(String filePath) {
-        this.filePath = filePath;
+        super(filePath);
     }
     
     /**
      * Parse the configured report card file
      * 
+     * @param br BufferedReader to read the file
      * @return ReportCard object containing the data from the configured report card file
      */
     @Override
-    public ReportCard parse() {
-        // REPLACE ME WITH A REAL IMPLEMENTATION!
+    public ReportCard parse(BufferedReader br) {
+        try
+        {
+            //Declare Variables
+            String name = "", dob; //Student's name and date of birth
+            
+            String currentLine = br.readLine(); //Name line
+            name = currentLine;
+            
+            currentLine = br.readLine(); //Dare of Birth Line
+            dob = currentLine;
+            
+            br.readLine(); //Blank line
+            
+            currentLine = br.readLine(); //Marks line
+            currentLine = currentLine.replaceAll(",", " "); //Replace commas with spaces for tokenizer
```

Comment:
For future reference, the StringTokenizer class has a constructor which takes a delimiter, so you can have it tokenize based on commas rather than whitespace